### ISSUE TITLE
kernel hang in fchownat?

### ISSUE THREAD

**Issue by djs55** [linuxkit/linuxkit#1376]
A Docker for Mac user reported `dockerd` hanging in Moby on `17.03.0-ce-mac2`:  https://github.com/docker/for-mac/issues/1461 

The `dmesg` shows the `dockerd` process is hanging in `fchownat`:

```
"echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
dockerd         D    0  2369   1871 0x00000000
 00000000000188c0 0000000000000000 ffff920c77456c00 ffff920c79620180
 ffff920c7f941100 ffff920c7c8988c0 ffffffff8c84525e 0000000000000002
 ffff920c7f941100 0000000000000359 ffff920c709523c0 ffff920c7095c840
Call Trace:
 [<ffffffff8c84525e>] ? __schedule+0x35f/0x43d
 [<ffffffff8c8453ba>] ? schedule+0x7e/0x87
 [<ffffffff8c209b02>] ? d_alloc_parallel+0x1cf/0x2c8
 [<ffffffff8c1a072f>] ? __rmqueue_smallest+0x97/0xaf
 [<ffffffff8c0ff097>] ? wake_up_q+0x45/0x45
 [<ffffffff8c1feb8b>] ? lookup_slow+0x83/0x12d
 [<ffffffff8c2006c2>] ? walk_component+0x76/0x145
 [<ffffffff8c1fe814>] ? __inode_permission+0x43/0xae
 [<ffffffff8c20089f>] ? link_path_walk+0x10e/0x44a
 [<ffffffff8c200ce8>] ? path_init+0x10d/0x26e
 [<ffffffff8c200eee>] ? path_lookupat+0x31/0x104
 [<ffffffff8c2015ca>] ? filename_lookup+0x8c/0x11d
 [<ffffffff8c1fee13>] ? unlazy_walk+0x9f/0x119
 [<ffffffff8c1e14d3>] ? ___cache_free+0xd1/0x126
 [<ffffffff8c1f2d55>] ? __check_object_size+0x157/0x17c
 [<ffffffff8c4422bd>] ? strncpy_from_user+0x47/0xff
 [<ffffffff8c1f9ffd>] ? vfs_fstatat+0x5b/0xad
 [<ffffffff8c1f9ffd>] ? vfs_fstatat+0x5b/0xad
 [<ffffffff8c1fa2b4>] ? SYSC_newstat+0x20/0x4c
 [<ffffffff8c20ff56>] ? mntput_no_expire+0x18/0x122
 [<ffffffff8c1f3c6a>] ? SYSC_fchownat+0xaa/0xe1
 [<ffffffff8c847db7>] ? entry_SYSCALL_64_fastpath+0x1a/0xa9
INFO: task dockerd:2369 blocked for more than 120 seconds.
      Not tainted 4.9.12-moby #1
```
To my untrained eye it doesn't look FUSE-related so perhaps it's related to `overlay2`? The `df` looks fine in the VM (often a broken filesystem causes this to hang):
```
$ cat df\ -h.stdout 
Filesystem                Size      Used Available Use% Mounted on
tmpfs                   999.4M    153.8M    845.6M  15% /
tmpfs                   199.9M    220.0K    199.7M   0% /run
cgroup_root              10.0M         0     10.0M   0% /sys/fs/cgroup
dev                      10.0M         0     10.0M   0% /dev
shm                     999.4M         0    999.4M   0% /dev/shm
/dev/vda1                62.7G      1.4G     58.1G   2% /var
tmpfs                   199.9M         0    199.9M   0% /tmp
tmpfs                   999.4M    168.0K    999.2M   0% /Database
osxfs                   464.8G    209.3G    255.3G  45% /Users
osxfs                   464.8G    209.3G    255.3G  45% /host_docker_app
/dev/vda1                62.7G      1.4G     58.1G   2% /var/lib/docker/overlay2
overlay                  62.7G      1.4G     58.1G   2% /var/lib/docker/overlay2/42b3905b72044beaebc928d89c95040fbe4d5fb135d3093a8e26aa3a550b843c/merged
shm                      64.0M         0     64.0M   0% /var/lib/docker/containers/7d03dc61ff88e33f7b0c4a3844903401711b58d1c8f2854bdcb3391be78bce27/shm
```
I believe the Moby is built from 8dce12e433e6327e5175206ab1784610da0d141b

**Comment by rneugeba** [comment]
@djs55 @dsheets  is this still an issue? I also see there is now another issue linked. Just wondering if this one here can be closed in favour of the others?

**Comment by dsheets** [comment]
We don't know the cause of the issue and the deadlock is wholly within the VFS blocked on an FS tree lock. It would be good to have a log pattern created for this issue and checked against the corpus of Docker for Mac diagnostics.

**Comment by rn** [comment]
I don't think we have seen this for quite some time so closing. Please re-open if this still exists
